### PR TITLE
ElementR: Add `CryptoApi.getCrossSigningKeyId`

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * ElementR: Add `CryptoApi.getCrossSigningKeyId` ([\#3619](https://github.com/matrix-org/matrix-js-sdk/pull/3619)). Contributed by @florianduros.

### CODE REVIEW

File: src/crypto-api.ts [hunk 1]
```diff
@@ -479,6 +480,17 @@ export enum CrossSigningKey {
     UserSigning = "user_signing",
 }
 
+/**
+ * Cross signing key
+ * More information https://spec.matrix.org/v1.7/client-server-api/#post_matrixclientv3keysdevice_signingupload
```
Comment (richvdh) [inline src/crypto-api.ts:485]: image


```suggestion
 *
 * @see https://spec.matrix.org/v1.7/client-server-api/#post_matrixclientv3keysdevice_signingupload
```

**Review by richvdh — CHANGES_REQUESTED** [review]
There's slightly more to it than this. The api doc for `CryptoApi.getCrossSigningKeyId` says:

> If cross-signing has been initialised on this device, the ID of the given key. Otherwise, null

... which I think means we should call `OlmMachine.crossSigningStatus()` to check that it has been initialised.

And also we should probably call `OwnUserIdentity::is_verified` to check that the  public key matches the private key?

**Comment by florianduros** [comment]
> 



> There's slightly more to it than this. The api doc for `CryptoApi.getCrossSigningKeyId` says:
> 
> > If cross-signing has been initialised on this device, the ID of the given key. Otherwise, null
> 
> ... which I think means we should call `OlmMachine.crossSigningStatus()` to check that it has been initialised.
> 
> And also we should probably call `OwnUserIdentity::is_verified` to check that the public key matches the private key?
Oh, I thought that if we had an `UserIdentity`, it means that the cross signing has been initialized

File: src/rust-crypto/rust-crypto.ts [hunk 2]
```diff
@@ -422,8 +422,13 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
...
-        if (!userIdentity) {
+        const crossSigningStatus: RustSdkCryptoJs.CrossSigningStatus = await this.olmMachine.crossSigningStatus();
+        const privateKeysOnDevice =
+            crossSigningStatus.hasMaster && crossSigningStatus.hasUserSigning && crossSigningStatus.hasSelfSigning;
+
+        // The public or private keys are not available on this device
+        // Or the current user is not verified
+        if (!userIdentity?.isVerified() || !privateKeysOnDevice) {
             return null;
         }
```
Comment (richvdh) [inline src/rust-crypto/rust-crypto.ts]: This is a bit hard to follow
```suggestion
        if (!userIdentity || !privateKeysOnDevice) {
            // The public or private keys are not available on this device
            return null;
        }
        
        if (!userIdentity.isVerified()) {
            // We have both public and private keys, but they don't match!
            return null;
        }
```

File: spec/integ/crypto/cross-signing.spec.ts [hunk 3]
```diff
@@ -288,4 +288,55 @@ describe.each(Object.entries(CRYPTO_BACKENDS))("cross-signing (%s)", (backend: s
...
+                    // Override the routes define in `mockSetupCrossSigningRequests`
+                    { overwriteRoutes: true },
+                );
+            });
+        }
+
+        it("should return the cross signing key id for each cross signing key", async () => {
+            mockSetupCrossSigningRequests();
+
+            // Intercept cross singing keys upload
```
Comment (richvdh) [inline spec/integ/crypto/cross-signing.spec.ts:321]: :musical_note: 

```suggestion
            // Intercept cross signing keys upload
```

**Review by richvdh — APPROVED** [review]
It would be nice to have tests for these edge cases, but I guess it's fine for now.
